packages/react/src/components/Tag/Tag.js

```diff
@@ -77,12 +79,18 @@ const Tag = ({
       </button>
     </div>
   ) : (
-    <span
-      className={tagClasses}
-      title={typeof children === 'string' ? children : null}
-      {...other}>
-      {children !== null && children !== undefined ? children : TYPES[type]}
-    </span>
+    <div className={tagClasses} id={tagId} {...other}>
+      {CustomIconElement ? (
+        <div className={`${prefix}--tag__custom-icon`}>
+          <CustomIconElement />
+        </div>
+      ) : (
+        ''
+      )}
+      <span title={typeof children === 'string' ? children : null} {...other}>
```

Comment:
Noticed that `{...other}` was being spread to both the outer `<div>` and the `<span>` here, I think we'll need to pick one and stick with it otherwise we'll end up in weird situations like both are receiving an `onClick` or `data-testid` or something 🤔 

I think just keeping `{...other}` on the outermost node would work, but curious what you think!

Reply:
I agree, keeping it on the outermost node works better. Good catch, thanks @joshblack!